### PR TITLE
feat: List component

### PR DESCRIPTION
**Pull request by callinmullaney**
**Summary**
- Refactors and simplifies the basic list component

This PR fixes/implements the following **bugs/features**
- Rewrites the list component to be a simple but extendable base component. 

Explain the **motivation** for making this change. What existing problem does the pull request solve?
- Current list implementation element specific which complicates how molecules/organisms implement this component. 
- Utilizes add_attributes function for other components (like menus) to attach accessibility improvements as well as custom data- attributes.

**Documentation Update (required)**
- None

**How to review this PR**
- [ ] checkout branch and run `npm install` then `npm run develop`
- [  Go to `Atoms/Lists` and confirm you can see examples of order and unordered lists
- [ ] Inspect the markup and confirm it's using `bem` class structure
- [ ] Review the twig and styles and verify they are a good starting point for future projects and aren't too opinionated.

### CODE REVIEW

File: components/01-atoms/lists/_list.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ol {
+  @include reset;
+
+  list-style-type: decimal;
+
+  ol {
+    list-style-type: lower-alpha;
+
+    ol {
+      list-style-type: lower-roman;
```
Comment (ryanhagerty) [inline components/01-atoms/lists/_list.scss:30]: For the `ol` and `ul`, I don't think we should be making these type of decisions for `list-style-type` - seems like it's very context dependent. 

If this is for an example, maybe move that kind of code into the `stories` file? If not, feel free to let me know the reasoning.

**Review by ryanhagerty — CHANGES_REQUESTED** [review]
Couple comments but great work 🍟

File: components/01-atoms/lists/_list.scss [hunk 2]
```diff
@@ -0,0 +1,39 @@
+@mixin reset {
```
[the remark this answers is not in the document]
Comment (callinmullaney) [reply components/01-atoms/lists/_list.scss]: Yes it does and I removed this from the PR... I think that (eventually) these type of mixins should probably exist within the components that are relevant to them. But for now we can keep the `mixin.scss` file in place and rely on that structure until we implement support for the newer version of Sass.

File: components/01-atoms/lists/_list.scss (hunk 1 above)
Comment (callinmullaney) [reply components/01-atoms/lists/_list.scss]: I agree with you here. Removed.

File: components/01-atoms/lists/list.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+unordered__list__items:
+  - list__item__content: "This is a list of items that doesn't need numbers.<ul><li>They may have sub-levels.<ul><li>Here's another level.</li></ul></li></ul>"
```
[the remark this answers is not in the document]
Comment (callinmullaney) [reply components/01-atoms/lists/list.yml]: Agreed. I did this originally to show the different nested styles for list but given your feedback above I removed these list-style styles. I did explore implementing a macro that would loop through the yml file and auto-generate the nested lists but this approach seemed to over-complicate this simple list component and also require some menu refactoring.

**Comment by callinmullaney** [comment]
@ryanhagerty I agree with all the feedback you left me here and made these changes. Can you give this a quick once-over and confirm?

**Review by ryanhagerty — APPROVED** [review]
Branch just needs updating, other than that approvo! 🪄 
Good work Callin

**Pull request merged**